### PR TITLE
The History of Goal edits

### LINKED ISSUE #7: The History of Goal edits
- title/description edited
- goals connected/disconnected
- tags added/removed
- participants added/removed
- state changed

### PR DESCRIPTION
**Pull request by LamaEats**
New data table `GoalHistory` that storing any change of goal, ex. change state, change title/description, add/remove tags or participants, also storing previous and next values of subjects for each record

Also seed for first goal in project by `prisma migrate dev` command.

Database table scheme and migration

## PR includes

- [X] Feature

## Related issues

Resolve #7 

## QA Instructions, Screenshots, Recordings
  
image
image

### CODE REVIEW

File: prisma/seed.ts [hunk 1]
```diff
@@ -300,6 +300,24 @@ seed('Default projects', async () => {
...
+            await prisma.goalHistory.create({
+                data: {
+                    goalId: goal.id,
+                    subject: 'state',
+                    action: 'change',
+                    previousValue: 'InProgress',
+                    nextValue: 'Critical',
+                    activityId: sample(allUsers).activityId,
+                },
+                include: {
```
Comment (awinogradov) [inline prisma/seed.ts:316]: Unused values. `include` makes sense if you are using result of req, but here you don't.

File: trpc/router/goalHistory.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { TRPCError } from '@trpc/server';
```
Comment (awinogradov) [inline trpc/router/goalHistory.ts:1]: I'm not sure we really need this endpoint. We will use prisma methods in other routes.

**Review by awinogradov — CHANGES_REQUESTED** [review]
- remove `router/goalHistory`
- remove unused value in seed.ts

File: prisma/seed.ts [hunk 3]
```diff
@@ -300,6 +300,20 @@ seed('Default projects', async () => {
...
+
+            const [goal] = allGoals;
+
+            // eslint-disable-next-line no-await-in-loop
+            await prisma.goalHistory.create({
+                data: {
+                    goalId: goal.id,
+                    subject: 'state',
+                    action: 'change',
+                    previousValue: 'InProgress',
```
Comment (awinogradov) [inline prisma/seed.ts:312]: We need to write state.id here. If we want to view some meta about states later.

File: trpc/queries/goals.ts [hunk 4]
```diff
@@ -302,6 +310,7 @@ export const addCalclulatedGoalsFields = (goal: any, activityId: string) => {
         _isEditable,
         _lastEstimate,
         _shortId,
+        history,
```
Comment (awinogradov) [inline trpc/queries/goals.ts:313]: Add underscore to mark field as calculated

File: trpc/router/goal.ts [hunk 5]
```diff
@@ -392,6 +463,15 @@ export const goal = router({
                     data: {
                         id, // this is hack to force updatedAt field
                         [kind]: { [connectionMap[String(direction)]]: connection },
+                        history: {
+                            create: {
+                                activityId: ctx.session.user.activityId,
+                                subject: 'dependencies',
+                                action: direction ? 'add' : 'remove',
+                                // FIXME: scoped id of deps goal
```
Comment (awinogradov) [inline trpc/router/goal.ts:471]: It's not required. We are using shortId just for view and user navigation

**Review by awinogradov — APPROVED** [review]

File: trpc/queries/goals.ts (hunk 4 above)
Comment (LamaEats) [reply trpc/queries/goals.ts:313]: this is unused field, will drop it

File: trpc/router/goal.ts (hunk 5 above)
Comment (LamaEats) [reply trpc/router/goal.ts:471]: i'm forgot drop this comment :)

**Pull request merged**